### PR TITLE
convert-thumbnail switch added

### PR DESCRIPTION
**Pull request by mostafa-norouzi**
Through this, you can force yt-dlp to make preferred image format other than default :

`->convertThumbnail("png")
`

### CODE REVIEW

File: src/Options.php [hunk 1]
```diff
@@ -150,6 +150,8 @@ class Options
     // Thumbnail Images Options
     private bool $writeThumbnail = false;
     private bool $writeAllThumbnails = false;
+    private string $convertThumbnail = 'jpg';
```
Comment (norkunas) [inline src/Options.php:153]: I think by default it shouldn't be set at all?

File: src/Options.php [hunk 2]
```diff
@@ -150,6 +150,8 @@ class Options
     // Thumbnail Images Options
     private bool $writeThumbnail = false;
     private bool $writeAllThumbnails = false;
+    private string $convertThumbnail ;
```
Comment (norkunas) [inline src/Options.php:153]:
```suggestion
    private string $convertThumbnail = null;
```

**Review by norkunas — CHANGES_REQUESTED** [review]

File: src/Options.php [hunk 3]
```diff
@@ -1023,6 +1025,17 @@ public function writeThumbnail(bool $writeThumbnail): self
         return $new;
     }
 
+    /**
+     * Convert thumbnail to another format.
+     */
+    public function convertThumbnail(string $format): self
```
Comment (norkunas) [inline src/Options.php:1031]: Is there are list of accepted formats so we could typehint against them?

File: src/Options.php [hunk 4]
```diff
@@ -150,7 +150,7 @@ class Options
     // Thumbnail Images Options
     private bool $writeThumbnail = false;
     private bool $writeAllThumbnails = false;
-    private string $convertThumbnail ;
+    private string $convertThumbnail = null ;
```
Comment (norkunas) [inline src/Options.php:153]: I don't understand why you add a space after null, you need to fix CS

Comment (norkunas) [inline src/Options.php:153]: also make argument nullable

Comment (mostafa-norouzi) [reply src/Options.php:153]: Sorry, this is my first time to contribute something and environment is new for me :)

Comment (norkunas) [reply src/Options.php]: You can run locally `vendor/bin/php-cs-fixer fix` :)

File: src/Options.php [hunk 5]
```diff
@@ -1023,6 +1025,18 @@ public function writeThumbnail(bool $writeThumbnail): self
         return $new;
     }
 
+    /**
+     * Convert thumbnail to another format.
+     * supported: jpg, png, webp
```
Comment (norkunas) [inline src/Options.php:1030]:
```suggestion
     *
     * @param 'jpg'|'png'|'webp'|null $format
```

File: src/Options.php (hunk 4 above)
Comment (mostafa-norouzi) [reply src/Options.php]: I ran the php-cs-fixer with fix switch but still error here! PHP version is matter?

**Review by norkunas — APPROVED** [review]

**Comment by norkunas** [comment]
Thank you @mostafa-norouzi and congrats on your first contribution

**Comment by mostafa-norouzi** [comment]
Thank you for your patience! But at last I didn't find out what was the exact problem.

**Pull request merged**